freqtrade/exchange/exchange.py

```diff
@@ -559,7 +561,20 @@ def get_min_pair_stake_amount(self, pair: str, price: float,
         # The value returned should satisfy both limits: for amount (base currency) and
         # for cost (quote, stake currency), so max() is used here.
         # See also #2575 at github.
-        return max(min_stake_amounts) * amount_reserve_percent
+        return self.apply_leverage_to_stake_amount(
+            max(min_stake_amounts) * amount_reserve_percent,
+            leverage or 1.0
+        )
+
+    def apply_leverage_to_stake_amount(self, stake_amount: float, leverage: float):
+        """
+        # * Should be implemented by child classes if leverage affects the stake_amount
+        Takes the minimum stake amount for a pair with no leverage and returns the minimum
+        stake amount when leverage is considered
+        :param stake_amount: The stake amount for a pair before leverage is considered
+        :param leverage: The amount of leverage being used on the current trade
+        """
+        return stake_amount
```

Comment:
I think the combination of these 2 is a very dangerous premise.

While i understand both will be necessary (the latter one due to kraken)- i think we need to be careful what "stake_amount" represents in the freqtrade bot context - and be consistent throughout.

Will it be used as "stake_amount * leverage * price" - or will we always buy "stake_amount * price" - just when using leverage, it'll not cost us "stake_amount" - but "stake-amount / leverage".

Reply:
I understand the minimum stake_amount as the minimum amount of currency that you can put forward in order to make a trade, if the trade costs you less than the minimum amount that you must put forward, we still need to know what that minimum amount is so that we can make sure that we put in more than that amount

Reply:
so basically - stake_amount is the amount that i need to buy in order to get into the trade (with or without leverage).

I'm just not certain so far if the 2 will collide (it'll also depend how they're going to be used).

If i wanna buy on kraken for 15€ - it can cost me 15€ (no leverage) - or it can cost me 7.5€ (2x).
assumption (and usage) for `get_min_stake_amount` is to give you 15€ - as that's the mininmum amount the exchange will allow you to trade.

On binance however, if i wanna get in for 15$ - i can pay 15$ and get coins worth 15$ (1x) - or i can pay 15$ - and get coins worth 30$ (2x).
Would binance allow me to only pay 7.5$ to get 15$ with 2x leverage? 

so while the above comment mentions "apply leverage to stake" - it's unclear how implementation of this should look like - and i think it should be clear from the docstring, so future implementers of new exchanges can do so without issues.

Reply:
> Would binance allow me to only pay 7.5$ to get 15$ with 2x leverage?

Yeah, the binance class overrides this method with that logic

Reply:
The default is to just use the regular stake amount, but if leverage lets you lower that amount, then the exchange overrides the method

Reply:
so the expectation is that kraken will not override / change this method, while binance will (this is just for my understanding).